src/teachy_tv.c

```diff
@@ -0,0 +1,1674 @@
+    LZDecompressWram(&gUnknown_8E86D6C, gUnknown_203F450->buffer4);
+    LoadCompressedPalette(&gUnknown_8E86F98, 0, 0x80);
+    LoadPalette(&src, 0, 2);
+    LoadSpritePalette(&gUnknown_83A5348);
+    TeachyTvLoadBg3Map(gUnknown_203F450->buffer3);
+}
+
+void TeachyTvCreateAndRenderRbox()
+{
+    InitWindows(&gUnknown_84792F0);
+    DeactivateAllTextPrinters();
+    FillWindowPixelBuffer(0, 0xCC);
+    PutWindowTilemap(0);
+    PutWindowTilemap(1);
+    CopyWindowToVram(0, 2);
+}
+
+u8 TeachyTvSetupWindow()
+{
+    int hasItem;
+    gMultiuseListMenuTemplate = gUnknown_8479368;
+    gMultiuseListMenuTemplate.windowId = 1;
+    gMultiuseListMenuTemplate.moveCursorFunc = TeachyTvAudioByInput;
+    if (!CheckBagHasItem(ITEM_TM_CASE, 1))
+    {
+        gMultiuseListMenuTemplate.items = &gUnknown_8479340;
+        gMultiuseListMenuTemplate.totalItems = 5;
+        gMultiuseListMenuTemplate.maxShowed = 5;
+        gMultiuseListMenuTemplate.upText_Y = (gMultiuseListMenuTemplate.upText_Y + 8) & 0xF;
+    }
+    return ListMenuInit(
+               &gMultiuseListMenuTemplate,
+               gTeachyTV_StaticResources.scrollOffset,
+               gTeachyTV_StaticResources.selectedRow);
+}
+
+void TeachyTvSetupScrollIndicatorArrowPair()
+{
+    int hasItem;
+    if (!CheckBagHasItem(ITEM_TM_CASE, 1))
+    {
+        struct TeachyTvBuf * temp = gUnknown_203F450;
+        temp->var_4007 = 0xFF;
+    }
+
+    else {
+        struct TeachyTvBuf *temp; 
+        u8 res;
+        res = AddScrollIndicatorArrowPair(&gUnknown_8479380, &(gTeachyTV_StaticResources.scrollOffset));
+        temp = gUnknown_203F450;
+        temp->var_4007 = res;
+        }
+}
+
+void TeachyTvRemoveScrollIndicatorArrowPair()
+{
+    if ( gUnknown_203F450->var_4007 != 0xFF )
+    {
+        RemoveScrollIndicatorArrowPair(gUnknown_203F450->var_4007);
+        gUnknown_203F450->var_4007 = 0xFF;
+    }
+}
+
+void TeachyTvAudioByInput(s32 notUsed, bool8 play, struct ListMenu *notUsedAlt)
+{
+    if ( play != TRUE )
+        PlaySE(SE_SELECT);
+}
+
+void TeachyTvInitIo()
+{
+    SetGpuReg(REG_OFFSET_WININ, 0x3F);
+    SetGpuReg(REG_OFFSET_WINOUT, 0x1F);
+    SetGpuReg(REG_OFFSET_BLDCNT, 0xCC);
+    SetGpuReg(REG_OFFSET_BLDY, 0x5);
+}
+
+u8 TeachyTvSetupObjEventAndOam()
+{
+    u8 temp = AddPseudoEventObject(90, SpriteCallbackDummy, 0, 0, 8);
+    gSprites[temp].oam.priority = 2;
+    gSprites[temp].invisible = 1;
+    return temp;
+}
+
+void TeachyTvSetSpriteCoordsAndSwitchFrame(u8 objId, u16 x, u16 y, u8 frame)
+{
+    gSprites[objId].pos2.x = x;
+    gSprites[objId].pos2.y = y;
+    gSprites[objId].invisible = 0;
+    StartSpriteAnim(&gSprites[objId], frame);
+}
+
+void TeachyTvSetWindowRegs()
+{
+    SetGpuReg(REG_OFFSET_WIN0V, 0xC64);
+    SetGpuReg(REG_OFFSET_WIN0H, 0x1CD4);
+}
+
+void TeachyTvClearWindowRegs()
+{
+    SetGpuReg(REG_OFFSET_WIN0V, 0x0);
+    SetGpuReg(REG_OFFSET_WIN0H, 0x0);
+}
+
+void TeachyTvBg2AnimController()
```

Comment:
If no arguments, declare it as `void foo(void)`

Reply:
```c

void TeachyTvBg2AnimController(void)
{
    u16 * tilemapBuffer = GetBgTilemapBuffer(2);
    u8 i;
    u8 j;
    for (i = 1; i < 13; i++)
    {
        for (j = 2; j < 25; j++)
        {
            int x = Random();
            tilemapBuffer[32 * i + j] = ((x &= 3) << 10); // gamefreak pls
            tilemapBuffer[32 * i + j] += 0x301F;
        }
    }
    schedule_bg_copy_tilemap_to_vram(2);
}
```

Reply:
> ```c
> void TeachyTvBg2AnimController(void)
> {
>     u16 * tilemapBuffer = GetBgTilemapBuffer(2);
>     u8 i;
>     u8 j;
>     for (i = 1; i < 13; i++)
>     {
>         for (j = 2; j < 25; j++)
>         {
>             int x = Random();
>             tilemapBuffer[32 * i + j] = ((x &= 3) << 10) + 0x301F; // gamefreak pls
>         }
>     }
>     schedule_bg_copy_tilemap_to_vram(2);
> }
> ```

didn't work, will examine the cause later